### PR TITLE
Fixes for routes

### PR DESCRIPTION
**Pull request by teelu**
Please review this PR First.  Quick fixes and reordering of when I update status

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -84,20 +84,19 @@ def car_pick_up
...
     @request = Request.find_by!('valet_pick_up_id = ? AND id = ?', Valet.find(params[:valet_id]).id, params[:request_id])
     @request.update(bay_number: valet_car_parked_params[:bay_number])
-    @request.keys_dropped!
     parking_spot = {
       latitude:   @request.parking_location.latitude,
       longitude:  @request.parking_location.longitude,
       address:    @request.parking_location.address
     }
     PrivatePub.publish_to "/user/#{@request.id}", :parking_spot => parking_spot 
+    @request.keys_dropped!
```
Comment (h8rry) [inline app/controllers/requests_controller.rb:93]: Why move everything down?

Comment (teelu) [inline app/controllers/requests_controller.rb:93]: was some strange issue where it updated too early sometimes

Comment (h8rry) [inline app/controllers/requests_controller.rb:93]: Did you figure out what the problem was?

Comment (teelu) [inline app/controllers/requests_controller.rb:93]: Nope, but it works when i moved it down.  I think I was just updating the status too early before I validate the update actually goes through?  So far i haven't had an issue

**Pull request merged**